// @guardian/libs/dist/cjs/index.js
"use strict";
/* istanbul ignore file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.debug = exports.log = exports.storage = exports.loadScript = exports.isUndefined = exports.isString = exports.getLocale = void 0;
var getLocale_1 = require("./getLocale");
Object.defineProperty(exports, "getLocale", { enumerable: true, get: function () { return getLocale_1.getLocale; } });
var isString_1 = require("./isString");
Object.defineProperty(exports, "isString", { enumerable: true, get: function () { return isString_1.isString; } });
var isUndefined_1 = require("./isUndefined");
Object.defineProperty(exports, "isUndefined", { enumerable: true, get: function () { return isUndefined_1.isUndefined; } });
var loadScript_1 = require("./loadScript");
Object.defineProperty(exports, "loadScript", { enumerable: true, get: function () { return loadScript_1.loadScript; } });
var storage_1 = require("./storage");
Object.defineProperty(exports, "storage", { enumerable: true, get: function () { return storage_1.storage; } });
var logger_1 = require("./logger");
Object.defineProperty(exports, "log", { enumerable: true, get: function () { return logger_1.log; } });
Object.defineProperty(exports, "debug", { enumerable: true, get: function () { return logger_1.debug; } });
